server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -5619,7 +5619,7 @@ public VirtualMachine migrateVirtualMachine(Long vmId, Host destinationHost) thr
             throw new InvalidParameterValueException("Unsupported Hypervisor Type for User VM migration, we support XenServer/VMware/KVM/Ovm/Hyperv/Ovm3 only");
         }
 
-        if (isVMUsingLocalStorage(vm)) {
+        if (isVMUsingLocalStorage(vm.getId())) {
             if (s_logger.isDebugEnabled()) {
                 s_logger.debug(vm + " is using Local Storage, cannot migrate this VM.");
```

Comment:
Is this debug line still valid?

Reply:
That is a good point. It depends on the strategy configured at the ResourceService, I will take a look at the best approaches to adapt it.

Reply:
@wido I did check it again and noticed that the log message as it is makes sense.

The `migrateVirtualMachine` method does not work with volume migration. When a VM with volumes, e.g. when migrating VMs with local storage, is migrated it is used `migrateWithStorage` method. Therefore, we need to keep it that way.

Reply:
@wido I think that I answered the _wrong_ question on previous comments ...

Regarding the debug line, personally, I would remove it. The idea behind it would be to avoid costly String operations and therefore optimize CloudStack processing. However, I think that there is a little optimization, and most of the cases CloudStack already runs in Debug mode.

I think that I saw @DaanHoogland and @rafaelweingartner discussing about it on the past.

Reply:
1. As for the message, looks like an extra check for the strategy is in order, but the flow might already guarantee it's validity.
2. As for the second interpretation of the question, as a matter of principle it is always good to check the logging level as parameters are always evaluated before passing and you never know how often the code is being executed.
3. As for the level, this is an obvious mis-use of debug. An error or at least warning or info situation is occurring here; the user might expect a migration but there is something impairing it. This is not a message for a developer, hence not a debug level message.

my €0,03 ;)

Reply:
@DaanHoogland your €0,03 is very appreciated ;)
I will change the log level from debug to error (maybe warn, but it looks more like an error).

Reply:
@DaanHoogland I changed the log level from debug to error on those cases.